open-craft/opencraft
# BB-2085: Add custom domain support

This adds a custom domain workflow into the new registration form & APIs.

**Testing instructions:**
1. Checkout this branch locally.
2. Start Ocim (inside vagrant `make dev.run`) 
3. Install the updated API driver: from `frontend/`, run `./scripts/build-api-client.sh`.
4. Run the frontend `npm start`.
5. Make a normal betatest application with an internal domain, click on the activation links and check that the instance domain are correctly provisioned.
6. Clear your local storage to avoid stored token issues (specifically, delete `access_token` and `refresh_token`).
7. Start a new beta test registration, but click on the "I want to use my own domain" link in the first page.
8. Click on the activation links and check that the instance has the correct domain names set.

**Reviewer:**

**Notes:** 
I currently fill the subdomain as a slugified domain + some random characters in the frontend to avoid clashes (if the domain is `test.com`, the subdomain will be `testcom-23m4`). Let me know if you want me to move that to the backend (probably into the serializer validator).
